eval/mt_bench_pipeline.py

```diff
@@ -0,0 +1,145 @@
+# type: ignore
+# pylint: disable=no-value-for-parameter,import-outside-toplevel,import-error
+from kfp import dsl, compiler, kubernetes
+from kfp.dsl import component, pipeline, Input, Output, Artifact, Model, importer
+from kfp.kubernetes import use_config_map_as_env, use_secret_as_env
+
+EVAL_IMAGE = "quay.io/sallyom/instructlab-ocp:eval"
+TOOLBOX_IMAGE = "registry.access.redhat.com/ubi9/toolbox"
+CANDIDATE_S3_URI = "s3://sallyom-eval-e58df6b0-606b-4749-96a5-a105657cb068/models/instructlab/granite-7b-lab"
+K8S_NAME = "kfp-model-server"
+
+# TODO: This is a placeholder for launching vLLM with candidate model
+@component(base_image=EVAL_IMAGE)
+def serve_candidate_model(
+    candidate_model: Input[Model],
+    candidate_server_url: Output[str],
+):
+    import os
+
+    model_name = os.path.basename(os.path.normpath(candidate_model.path))
+    os.environ["CANDIDATE_MODEL_NAME"] = model_name
+    print(f"Candidate Model: {model_name}")
+
+    # TODO: Replace this with the candidate model endpoint
+    # Launch_server call with https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L301C1-L327C1
+    candidate_server_url = os.getenv("JUDGE_MODEL_ENDPOINT")
+
+@component(base_image=EVAL_IMAGE)
+def run_mt_bench(
+    candidate_model: Input[Model],
+    candidate_server_url: Input[str],
+    mt_bench_output: Output[Artifact],
+    max_workers: str,
+    merge_system_user_message: bool,
+):
+    import os
+    import torch
+    from instructlab.eval.mt_bench import MTBenchEvaluator
+
+    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+
+    gpu_available = torch.cuda.is_available()
+    gpu_name = torch.cuda.get_device_name(torch.cuda.current_device()) if gpu_available else "No GPU available"
+
+    print(f"GPU Available: {gpu_available}, Using: {gpu_name}")
+
+    effective_device = device if device != "None" else ("cuda" if gpu_available else "cpu")
```

Comment:
Where the `device` variable comes from? I can't see it, am I blind?

Also, you can use `None` type, instead of `"None"` string. That would be better I think.

Reply:
updated.

Reply:
I'm confused. Did you forget to push the latest changes or you didn't update it?

Reply:
must've not pushed... 👀

Reply:
updated, pushed, done